src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -0,0 +1,497 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Recommendations;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Recommendations
+{
+    internal class CSharpRecommendationServiceRunner : AbstractRecommendationServiceRunner<CSharpSyntaxContext>
+    {
+        public CSharpRecommendationServiceRunner(
+            CSharpSyntaxContext context, bool filterOutOfScopeLocals, CancellationToken cancellationToken)
+            : base(context, filterOutOfScopeLocals, cancellationToken)
+        {
+        }
+
+        public override ImmutableArray<ISymbol> GetSymbols()
```

Comment:
i'm going to assume most of this move relatively unscathed over.  IF you did change one significantly, can you let me know?

Reply:
There are two changes:
1. Moving methods to runners classes to keep `context`, `filterOutOfScopeLocals `and `cancellationToken` in passed to the class. 
2. Moving 
`GetSymbols(INamespaceOrTypeSymbol container, int position, bool excludeInstance, bool useBaseReferenceAccessibility)`
`LookupSymbolsInContainer(INamespaceOrTypeSymbol container, int position, bool excludeInstance)
`
`
SuppressDefaultTupleElements(
            INamespaceOrTypeSymbol container, ImmutableArray<ISymbol> symbols)
`
to the base class.

Runners classes are used for executing one public method only. Currently, we pass all parameters from one static method to another. I need to make some methods to be instance methods for `ThenInclude` to simplify their overrides between C# and VB.

Reply:
Ok.  so this is basically the [Replace method with method object](https://refactoring.guru/replace-method-with-method-object) refactoring then?  Functionality otherwise stays the same?

Reply:
yes, basically it is it.